packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4251,6 +4259,12 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   @override
   InheritedWidget dependOnInheritedElement(InheritedElement ancestor, { Object? aspect }) {
     assert(ancestor != null);
+    assert(
+       !ancestor.clearDependencyOnRebuild || debugDoingBuild || !dirty,
+      'Cannot depend on an InheritedElement with clearDependencyOnRebuild: true '
+      ' in Widget life-cycles other than `build`.',
```

Comment:
nit: remove the space before "in" to avoid a double space in the string (maybe even just put the entire sting on one line.

Reply:
Similar to the error you get when trying to establish a dependency in `initState`, this should probably throw a more detailed FlutterError to guide the developer.